src/Cli/dotnet/Parser.cs

```diff
@@ -178,17 +178,23 @@ internal static int ExceptionHandler(Exception exception, ParseResult parseResul
 
             if (exception is Utils.GracefulException)
             {
-                Reporter.Error.WriteLine(CommandLoggingContext.IsVerbose
-                    ? exception.ToString().Red().Bold()
-                    : exception.Message.Red().Bold());
+                Reporter.Error.WriteLine(CommandLoggingContext.IsVerbose ?
+                    exception.ToString().Red().Bold() :
+                    exception.Message.Red().Bold());
             }
             else if (exception is CommandParsingException)
             {
-                Reporter.Error.WriteLine(CommandLoggingContext.IsVerbose
-                    ? exception.ToString().Red().Bold()
-                    : exception.Message.Red().Bold());
+                Reporter.Error.WriteLine(CommandLoggingContext.IsVerbose ?
+                    exception.ToString().Red().Bold() :
+                    exception.Message.Red().Bold());
                 parseResult.ShowHelp();
             }
+            else if (exception.GetType().Name.Equals("WorkloadManifestCompositionException"))
+            {
+                Reporter.Error.WriteLine(CommandLoggingContext.IsVerbose ?
+                    exception.ToString().Red().Bold() :
+                    exception.Message.Red().Bold());
+            }
             else
             {
                 Reporter.Error.Write("Unhandled exception: ".Red().Bold());
```

Comment:
Shouldn't we also apply the verbosity to the `else` to be consistent?

Reply:
Not unreasonable. I was focused on my particular exception, but I can extend it to the else.

Reply:
Thanks for taking a peek